cpp/include/cuspatial/soa_readers.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/cudf.h>
+#include <cuspatial/cuspatial.h>
+
+namespace cuspatial {
+
+	/**
+	*@Brief read uint32_t (unsigned integer with 32 bit fixed length) data from file as column;
+    *mostly for identifiers, lengths and positions.
+    *@param[in] col_fn: file name of uint32_t column data
+    *@param[out] ids: gdf_column storing the uint32_t column
+	*/
+	void read_uint_soa(const char *col_fn, gdf_column& ids);
```

Comment:
Usually in APIs that are processing a lot of data, parameter passing is not a bottleneck. :)

- Pass by value whenever it is inexpensive to do so. 
 - Use const reference parameters when passing an object that is too expensive to pass by value (primitive types such as your int example and simple structs are NOT expensive to pass by value). 
 - Prefer returning results rather than using "in/out" parameters.
 - If you must use an in/out parameter, pass it by a pointer rather than a non-const reference.  This is a matter of style, but it's a style we prefer in cuDF/RAPIDS because we believe it makes it more clear that this parameter's contents will be modified. A non-const reference doesn't make that clear.